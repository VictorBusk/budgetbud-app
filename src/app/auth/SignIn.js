import React, {Component} from 'react';
import {
    KeyboardAvoidingView,
    Keyboard,
    TouchableWithoutFeedback, View
} from 'react-native';
import {connect} from 'react-redux';
import {Container, Button, Label} from 'native-base';
import {AuthForm, Logo} from '../../components/index';
import {container, button, color} from "../../style/index";
import I18n from "../../strings/i18n";
import {
    codeChanged,
    cprNumberChanged,
    resetAuthState,
    signIn,
    resetAuthError
} from "../../redux/actions";
import {showWarningToast} from "../../helpers/toasts";

class SignIn extends Component {
    componentWillReceiveProps(nextProps) {
        if (nextProps.authError) {
            showWarningToast(nextProps.authError);
            this.props.resetAuthError();
        }
    }

    onCprNumberChange = (text) => {
        this.props.cprNumberChanged(text);
    };

    onCodeChange = (text) => {
        this.props.codeChanged(text);
    };

    handleSubmit = () => {
        Keyboard.dismiss();
        const {cprNumber, code} = this.props;
        this.props.signIn({cprNumber, code});
    };

    onForgotCodeButtonPress = () => {
        if (!this.props.authLoading) {
            this.props.resetAuthState(() => {
                this.props.navigation.navigate('RequestActivationCode');
            });
        }
    };

    onGoToSignUpButtonPress = () => {
        if (!this.props.authLoading) {
            this.props.resetAuthState(() => {
                this.props.navigation.navigate('SignUp');
            });
        }
    };

    render() {
        return (
            <KeyboardAvoidingView behavior='padding' style={container.signedOutContainer}>
                <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
                    <Container style={{alignSelf: 'stretch'}}>

                        <View style={[container.parentContainer, {flex: 2}]}>
                            <Logo logoText={I18n.t('signInLogoWelcome')}/>
                        </View>

                        <AuthForm handleSubmit={this.handleSubmit}
                                  onCprNumberChange={this.onCprNumberChange}
                                  onSecondInputChange={this.onCodeChange}
                                  cprNumber={this.props.cprNumber}
                                  secondInput={this.props.code}
                                  authLoading={this.props.authLoading}
                                  isSignIn={true}
                        />

                        <Container>
                            <Container style={container.optionContainer}>
                                <Button transparent style={button.optionButton}
                                        onPress={() => this.onForgotCodeButtonPress()}>
                                    <Label style={color.optionButton}>{I18n.t('signInForgotCode')}</Label>
                                </Button>

                                <Button transparent style={button.optionButton}
                                        onPress={() => this.onGoToSignUpButtonPress()}>
                                    <Label style={color.optionButton}>{I18n.t('signInGoToSignUp')}</Label>
                                </Button>
                            </Container>
                        </Container>

                    </Container>
                </TouchableWithoutFeedback>
            </KeyboardAvoidingView>
        );
    };
}

const mapStateToProps = ({auth}) => {
    return {cprNumber, code, authError, authLoading} = auth;
};

const mapDispatchToProps = {
    signIn,
    cprNumberChanged,
    codeChanged,
    resetAuthState,
    resetAuthError
};

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
