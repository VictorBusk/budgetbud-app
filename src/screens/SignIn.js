import React, {Component} from 'react';
import {
    KeyboardAvoidingView,
    Keyboard,
    TouchableWithoutFeedback, View
} from 'react-native';
import {connect} from 'react-redux';
import {Container, Button, Label} from 'native-base';
import {AuthForm, ErrorInfo, Logo} from '../components/';
import {
    signIn,
    cprNumberChanged,
    codeChanged,
    resetAuthState
} from '../actions';
import {container, button, color} from "../style";
import I18n from "../strings/i18n";

class SignIn extends Component {
    onCprNumberChange = (text) => {
        this.props.cprNumberChanged(text);
    };

    onCodeChange = (text) => {
        this.props.codeChanged(text);
    };

    handleSubmit = () => {
        Keyboard.dismiss();
        const {cprNumber, code} = this.props;
        this.props.signIn({cprNumber, code});
    };

    onForgotCodeButtonPress = () => {
        this.props.resetAuthState(() => {
            this.props.navigation.navigate('SendActivationCode');
        });
    };

    onGoToSignUpButtonPress = () => {
        this.props.resetAuthState(() => {
            this.props.navigation.navigate('SignUp');
        });
    };

    render() {
        return (
            <KeyboardAvoidingView behavior='padding' style={container.signedOutContainer}>
                <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
                    <Container style={{alignSelf: 'stretch'}}>

                        <View style={[container.parentContainer, {flex: 2}]}>
                            <Logo logoText={I18n.t('signInLogoWelcome')}/>
                        </View>

                        <AuthForm handleSubmit={this.handleSubmit}
                                  onCprNumberChange={this.onCprNumberChange}
                                  onSecondInputChange={this.onCodeChange}
                                  cprNumber={this.props.cprNumber}
                                  secondInput={this.props.code}
                                  authLoading={this.props.authLoading}
                                  isSignIn={true}
                        />

                        <Container>
                            <Container style={container.optionContainer}>
                                <Button transparent style={button.optionButton}
                                        onPress={() => this.onForgotCodeButtonPress()}>
                                    <Label style={color.optionButton}>{I18n.t('signInForgotCode')}</Label>
                                </Button>

                                <Button transparent style={button.optionButton}
                                        onPress={() => this.onGoToSignUpButtonPress()}>
                                    <Label style={color.optionButton}>{I18n.t('signInGoToSignUp')}</Label>
                                </Button>
                            </Container>

                            <ErrorInfo error={this.props.authError}/>
                        </Container>

                    </Container>
                </TouchableWithoutFeedback>
            </KeyboardAvoidingView>
        );
    };
}

const mapStateToProps = ({auth}) => {
    return {cprNumber, code, authError, authLoading} = auth;
};

const mapDispatchToProps = {
    signIn, cprNumberChanged, codeChanged, resetAuthState
};

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
