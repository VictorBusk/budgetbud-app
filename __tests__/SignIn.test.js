import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SignIn from '../src/screens/SignIn';

Enzyme.configure({adapter: new Adapter()});

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
    auth: {
        cprNumber: '',
        code: '',
        error: '',
        loading: false
    }
};

describe('Testing SignIn', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <SignIn/>,
            {context: {store: mockStore(initialState)}},
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
