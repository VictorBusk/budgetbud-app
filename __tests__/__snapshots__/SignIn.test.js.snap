// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignIn renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    authScreenSwitched={[Function]}
    code=""
    codeChanged={[Function]}
    cprNumber=""
    cprNumberChanged={[Function]}
    error=""
    loading={false}
    signIn={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "behavior": "padding",
      "children": <TouchableWithoutFeedback
        accessible={false}
        onPress={[Function]}
      >
        <Styled(Container)
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
          virtual={undefined}
        >
          <Logo />
          <AuthForm
            cprNumber=""
            error=""
            handleSubmit={[Function]}
            isSignIn={true}
            loading={false}
            onCprNumberChange={[Function]}
            onSecondInputChange={[Function]}
            secondInput=""
          />
          <Styled(Container)
            virtual={undefined}
          >
            <Styled(Container)
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "center",
                  "paddingTop": 13,
                  "width": "80%",
                }
              }
              virtual={undefined}
            >
              <Styled(Button)
                onPress={[Function]}
                style={
                  Object {
                    "height": 30,
                    "paddingLeft": 5,
                  }
                }
                transparent={true}
                virtual={undefined}
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "rgba(255,255,255,0.6)",
                    }
                  }
                  virtual={undefined}
                >
                  Glemt pinkode?
                </Styled(Label)>
              </Styled(Button)>
              <Styled(Button)
                onPress={[Function]}
                style={
                  Object {
                    "height": 30,
                    "paddingLeft": 5,
                  }
                }
                transparent={true}
                virtual={undefined}
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "rgba(255,255,255,0.6)",
                    }
                  }
                  virtual={undefined}
                >
                  Ny bruger?
                </Styled(Label)>
              </Styled(Button)>
            </Styled(Container)>
            <ErrorInfo
              error=""
            />
          </Styled(Container)>
        </Styled(Container)>
      </TouchableWithoutFeedback>,
      "keyboardVerticalOffset": 0,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#166a97",
        "flex": 2,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": false,
        "children": <Styled(Container)
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
          virtual={undefined}
        >
          <Logo />
          <AuthForm
            cprNumber=""
            error=""
            handleSubmit={[Function]}
            isSignIn={true}
            loading={false}
            onCprNumberChange={[Function]}
            onSecondInputChange={[Function]}
            secondInput=""
          />
          <Styled(Container)
            virtual={undefined}
          >
            <Styled(Container)
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "center",
                  "paddingTop": 13,
                  "width": "80%",
                }
              }
              virtual={undefined}
            >
              <Styled(Button)
                onPress={[Function]}
                style={
                  Object {
                    "height": 30,
                    "paddingLeft": 5,
                  }
                }
                transparent={true}
                virtual={undefined}
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "rgba(255,255,255,0.6)",
                    }
                  }
                  virtual={undefined}
                >
                  Glemt pinkode?
                </Styled(Label)>
              </Styled(Button)>
              <Styled(Button)
                onPress={[Function]}
                style={
                  Object {
                    "height": 30,
                    "paddingLeft": 5,
                  }
                }
                transparent={true}
                virtual={undefined}
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "rgba(255,255,255,0.6)",
                    }
                  }
                  virtual={undefined}
                >
                  Ny bruger?
                </Styled(Label)>
              </Styled(Button)>
            </Styled(Container)>
            <ErrorInfo
              error=""
            />
          </Styled(Container)>
        </Styled(Container)>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Logo />,
            <AuthForm
              cprNumber=""
              error=""
              handleSubmit={[Function]}
              isSignIn={true}
              loading={false}
              onCprNumberChange={[Function]}
              onSecondInputChange={[Function]}
              secondInput=""
            />,
            <Styled(Container)
              virtual={undefined}
            >
              <Styled(Container)
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "paddingTop": 13,
                    "width": "80%",
                  }
                }
                virtual={undefined}
              >
                <Styled(Button)
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 30,
                      "paddingLeft": 5,
                    }
                  }
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "rgba(255,255,255,0.6)",
                      }
                    }
                    virtual={undefined}
                  >
                    Glemt pinkode?
                  </Styled(Label)>
                </Styled(Button)>
                <Styled(Button)
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 30,
                      "paddingLeft": 5,
                    }
                  }
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "rgba(255,255,255,0.6)",
                      }
                    }
                    virtual={undefined}
                  >
                    Ny bruger?
                  </Styled(Label)>
                </Styled(Button)>
              </Styled(Container)>
              <ErrorInfo
                error=""
              />
            </Styled(Container)>,
          ],
          "style": Object {
            "alignSelf": "stretch",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cprNumber": "",
              "error": "",
              "handleSubmit": [Function],
              "isSignIn": true,
              "loading": false,
              "onCprNumberChange": [Function],
              "onSecondInputChange": [Function],
              "secondInput": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(Container)
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "alignSelf": "center",
                      "paddingTop": 13,
                      "width": "80%",
                    }
                  }
                  virtual={undefined}
                >
                  <Styled(Button)
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 30,
                        "paddingLeft": 5,
                      }
                    }
                    transparent={true}
                    virtual={undefined}
                  >
                    <Styled(Label)
                      style={
                        Object {
                          "color": "rgba(255,255,255,0.6)",
                        }
                      }
                      virtual={undefined}
                    >
                      Glemt pinkode?
                    </Styled(Label)>
                  </Styled(Button)>
                  <Styled(Button)
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 30,
                        "paddingLeft": 5,
                      }
                    }
                    transparent={true}
                    virtual={undefined}
                  >
                    <Styled(Label)
                      style={
                        Object {
                          "color": "rgba(255,255,255,0.6)",
                        }
                      }
                      virtual={undefined}
                    >
                      Ny bruger?
                    </Styled(Label)>
                  </Styled(Button)>
                </Styled(Container)>,
                <ErrorInfo
                  error=""
                />,
              ],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Styled(Button)
                      onPress={[Function]}
                      style={
                        Object {
                          "height": 30,
                          "paddingLeft": 5,
                        }
                      }
                      transparent={true}
                      virtual={undefined}
                    >
                      <Styled(Label)
                        style={
                          Object {
                            "color": "rgba(255,255,255,0.6)",
                          }
                        }
                        virtual={undefined}
                      >
                        Glemt pinkode?
                      </Styled(Label)>
                    </Styled(Button)>,
                    <Styled(Button)
                      onPress={[Function]}
                      style={
                        Object {
                          "height": 30,
                          "paddingLeft": 5,
                        }
                      }
                      transparent={true}
                      virtual={undefined}
                    >
                      <Styled(Label)
                        style={
                          Object {
                            "color": "rgba(255,255,255,0.6)",
                          }
                        }
                        virtual={undefined}
                      >
                        Ny bruger?
                      </Styled(Label)>
                    </Styled(Button)>,
                  ],
                  "style": Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "paddingTop": 13,
                    "width": "80%",
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Styled(Label)
                        style={
                          Object {
                            "color": "rgba(255,255,255,0.6)",
                          }
                        }
                        virtual={undefined}
                      >
                        Glemt pinkode?
                      </Styled(Label)>,
                      "onPress": [Function],
                      "style": Object {
                        "height": 30,
                        "paddingLeft": 5,
                      },
                      "transparent": true,
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Glemt pinkode?",
                        "style": Object {
                          "color": "rgba(255,255,255,0.6)",
                        },
                        "virtual": undefined,
                      },
                      "ref": null,
                      "rendered": "Glemt pinkode?",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Styled(Label)
                        style={
                          Object {
                            "color": "rgba(255,255,255,0.6)",
                          }
                        }
                        virtual={undefined}
                      >
                        Ny bruger?
                      </Styled(Label)>,
                      "onPress": [Function],
                      "style": Object {
                        "height": 30,
                        "paddingLeft": 5,
                      },
                      "transparent": true,
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Ny bruger?",
                        "style": Object {
                          "color": "rgba(255,255,255,0.6)",
                        },
                        "virtual": undefined,
                      },
                      "ref": null,
                      "rendered": "Ny bruger?",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "error": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "behavior": "padding",
        "children": <TouchableWithoutFeedback
          accessible={false}
          onPress={[Function]}
        >
          <Styled(Container)
            style={
              Object {
                "alignSelf": "stretch",
              }
            }
            virtual={undefined}
          >
            <Logo />
            <AuthForm
              cprNumber=""
              error=""
              handleSubmit={[Function]}
              isSignIn={true}
              loading={false}
              onCprNumberChange={[Function]}
              onSecondInputChange={[Function]}
              secondInput=""
            />
            <Styled(Container)
              virtual={undefined}
            >
              <Styled(Container)
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "paddingTop": 13,
                    "width": "80%",
                  }
                }
                virtual={undefined}
              >
                <Styled(Button)
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 30,
                      "paddingLeft": 5,
                    }
                  }
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "rgba(255,255,255,0.6)",
                      }
                    }
                    virtual={undefined}
                  >
                    Glemt pinkode?
                  </Styled(Label)>
                </Styled(Button)>
                <Styled(Button)
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 30,
                      "paddingLeft": 5,
                    }
                  }
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "rgba(255,255,255,0.6)",
                      }
                    }
                    virtual={undefined}
                  >
                    Ny bruger?
                  </Styled(Label)>
                </Styled(Button)>
              </Styled(Container)>
              <ErrorInfo
                error=""
              />
            </Styled(Container)>
          </Styled(Container)>
        </TouchableWithoutFeedback>,
        "keyboardVerticalOffset": 0,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#166a97",
          "flex": 2,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": false,
          "children": <Styled(Container)
            style={
              Object {
                "alignSelf": "stretch",
              }
            }
            virtual={undefined}
          >
            <Logo />
            <AuthForm
              cprNumber=""
              error=""
              handleSubmit={[Function]}
              isSignIn={true}
              loading={false}
              onCprNumberChange={[Function]}
              onSecondInputChange={[Function]}
              secondInput=""
            />
            <Styled(Container)
              virtual={undefined}
            >
              <Styled(Container)
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "paddingTop": 13,
                    "width": "80%",
                  }
                }
                virtual={undefined}
              >
                <Styled(Button)
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 30,
                      "paddingLeft": 5,
                    }
                  }
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "rgba(255,255,255,0.6)",
                      }
                    }
                    virtual={undefined}
                  >
                    Glemt pinkode?
                  </Styled(Label)>
                </Styled(Button)>
                <Styled(Button)
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 30,
                      "paddingLeft": 5,
                    }
                  }
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "rgba(255,255,255,0.6)",
                      }
                    }
                    virtual={undefined}
                  >
                    Ny bruger?
                  </Styled(Label)>
                </Styled(Button)>
              </Styled(Container)>
              <ErrorInfo
                error=""
              />
            </Styled(Container)>
          </Styled(Container)>,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Logo />,
              <AuthForm
                cprNumber=""
                error=""
                handleSubmit={[Function]}
                isSignIn={true}
                loading={false}
                onCprNumberChange={[Function]}
                onSecondInputChange={[Function]}
                secondInput=""
              />,
              <Styled(Container)
                virtual={undefined}
              >
                <Styled(Container)
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "alignSelf": "center",
                      "paddingTop": 13,
                      "width": "80%",
                    }
                  }
                  virtual={undefined}
                >
                  <Styled(Button)
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 30,
                        "paddingLeft": 5,
                      }
                    }
                    transparent={true}
                    virtual={undefined}
                  >
                    <Styled(Label)
                      style={
                        Object {
                          "color": "rgba(255,255,255,0.6)",
                        }
                      }
                      virtual={undefined}
                    >
                      Glemt pinkode?
                    </Styled(Label)>
                  </Styled(Button)>
                  <Styled(Button)
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 30,
                        "paddingLeft": 5,
                      }
                    }
                    transparent={true}
                    virtual={undefined}
                  >
                    <Styled(Label)
                      style={
                        Object {
                          "color": "rgba(255,255,255,0.6)",
                        }
                      }
                      virtual={undefined}
                    >
                      Ny bruger?
                    </Styled(Label)>
                  </Styled(Button)>
                </Styled(Container)>
                <ErrorInfo
                  error=""
                />
              </Styled(Container)>,
            ],
            "style": Object {
              "alignSelf": "stretch",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "cprNumber": "",
                "error": "",
                "handleSubmit": [Function],
                "isSignIn": true,
                "loading": false,
                "onCprNumberChange": [Function],
                "onSecondInputChange": [Function],
                "secondInput": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(Container)
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "alignSelf": "center",
                        "paddingTop": 13,
                        "width": "80%",
                      }
                    }
                    virtual={undefined}
                  >
                    <Styled(Button)
                      onPress={[Function]}
                      style={
                        Object {
                          "height": 30,
                          "paddingLeft": 5,
                        }
                      }
                      transparent={true}
                      virtual={undefined}
                    >
                      <Styled(Label)
                        style={
                          Object {
                            "color": "rgba(255,255,255,0.6)",
                          }
                        }
                        virtual={undefined}
                      >
                        Glemt pinkode?
                      </Styled(Label)>
                    </Styled(Button)>
                    <Styled(Button)
                      onPress={[Function]}
                      style={
                        Object {
                          "height": 30,
                          "paddingLeft": 5,
                        }
                      }
                      transparent={true}
                      virtual={undefined}
                    >
                      <Styled(Label)
                        style={
                          Object {
                            "color": "rgba(255,255,255,0.6)",
                          }
                        }
                        virtual={undefined}
                      >
                        Ny bruger?
                      </Styled(Label)>
                    </Styled(Button)>
                  </Styled(Container)>,
                  <ErrorInfo
                    error=""
                  />,
                ],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Styled(Button)
                        onPress={[Function]}
                        style={
                          Object {
                            "height": 30,
                            "paddingLeft": 5,
                          }
                        }
                        transparent={true}
                        virtual={undefined}
                      >
                        <Styled(Label)
                          style={
                            Object {
                              "color": "rgba(255,255,255,0.6)",
                            }
                          }
                          virtual={undefined}
                        >
                          Glemt pinkode?
                        </Styled(Label)>
                      </Styled(Button)>,
                      <Styled(Button)
                        onPress={[Function]}
                        style={
                          Object {
                            "height": 30,
                            "paddingLeft": 5,
                          }
                        }
                        transparent={true}
                        virtual={undefined}
                      >
                        <Styled(Label)
                          style={
                            Object {
                              "color": "rgba(255,255,255,0.6)",
                            }
                          }
                          virtual={undefined}
                        >
                          Ny bruger?
                        </Styled(Label)>
                      </Styled(Button)>,
                    ],
                    "style": Object {
                      "alignItems": "flex-start",
                      "alignSelf": "center",
                      "paddingTop": 13,
                      "width": "80%",
                    },
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <Styled(Label)
                          style={
                            Object {
                              "color": "rgba(255,255,255,0.6)",
                            }
                          }
                          virtual={undefined}
                        >
                          Glemt pinkode?
                        </Styled(Label)>,
                        "onPress": [Function],
                        "style": Object {
                          "height": 30,
                          "paddingLeft": 5,
                        },
                        "transparent": true,
                        "virtual": undefined,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Glemt pinkode?",
                          "style": Object {
                            "color": "rgba(255,255,255,0.6)",
                          },
                          "virtual": undefined,
                        },
                        "ref": null,
                        "rendered": "Glemt pinkode?",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <Styled(Label)
                          style={
                            Object {
                              "color": "rgba(255,255,255,0.6)",
                            }
                          }
                          virtual={undefined}
                        >
                          Ny bruger?
                        </Styled(Label)>,
                        "onPress": [Function],
                        "style": Object {
                          "height": 30,
                          "paddingLeft": 5,
                        },
                        "transparent": true,
                        "virtual": undefined,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Ny bruger?",
                          "style": Object {
                            "color": "rgba(255,255,255,0.6)",
                          },
                          "virtual": undefined,
                        },
                        "ref": null,
                        "rendered": "Ny bruger?",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "error": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
